src/Metal/MTLCompat.cs

```diff
@@ -1,9 +1,12 @@
-#if XAMCORE_2_0 || !MONOMAC
-
+using System;
 using ObjCRuntime;
 
 namespace Metal {
 
-}
+	public partial class MTLSharedTextureHandle {
+
+		[Obsolete ("Default constructor was removed. Do not use.")]
```

Comment:
there's no constructor in native ObjC

Use `[Obsolete ("Type is not meant to be created by user code.")]` since it will share the string with other similar cases.